src/role_scout.js

```diff
@@ -13,153 +13,219 @@ roles.scout.settings = {
+    creep.memory.skip.push(creep.memory.search.target);
+    delete creep.memory.scoutSkip;
+  } else if (creep.room.name === creep.memory.search.target) {
+    creep.memory.search.seen.push(creep.room.name);
+  }
+  !roles.scout.lookAround(creep, 0, false, true)
+  && !roles.scout.setNewTarget(creep)
+  && roles.scout.goFurther(creep);
+
+};
+
+roles.scout.move = function(creep) {
+  let costMatrix = function(roomName) {
+    if (!Memory.rooms[roomName] || (Memory.rooms[roomName].lastSeen < (Game.time - 5)))
+      return creep.room.getCostMatrixCallback(creep.targetPosObject, true, false, true)
+    return false;
+  }
 
-    if (creep.memory.last && creep.memory.last.pos3 && creep.pos.roomName !== creep.memory.last.pos3.roomName) {
-      creep.moveTo(25, 25);
-      return true;
+  let search = PathFinder.search(
+    creep.pos, {
+      pos: creep.targetPosObject,
+      range: 20
+    }, {
+      roomCallback: costMatrix
     }
+  );
 
-    if (creep.isStuck()) {
-      creep.moveRandom();
-      creep.say('ImStuck', true);
-      creep.log('Scout Stuck, Randomly Moving: ' + JSON.stringify(creep.memory.last) + ' ' + JSON.stringify(creep.isStuck()));
-      return true;
+  config.visualizer.enabled
+  && config.visualizer.showPathSearches
+  && visualizer.showSearch(search);
+
+  if (search.path.length === 0 || (creep.inBase() && creep.room.memory.misplacedSpawn)) {
+    if (creep.isStuck() && roles.scout.onBorder(creep)) {
+      creep.say('imstuck at the border', true);
+      if (config.scout.scoutSkipWhenStuck) {
+        creep.say('skipping', true);
+        creep.memory.scoutSkip = true;
+        delete creep.memory.last; // Delete to reset stuckness.
+      }
     }
 
-    if (search.path.length === 0 || (creep.inBase() && creep.room.memory.misplacedSpawn)) {
-      creep.say('hello', true);
-      //       creep.log(creep.pos + ' ' + targetPosObject + ' ' + JSON.stringify(search));
-      if (creep.isStuck() && onBorder(creep)) {
-        creep.say('imstuck at the border', true);
-        if (config.room.scoutSkipWhenStuck) {
-          creep.say('skipping', true);
-          creep.memory.scoutSkip = true;
-          delete creep.memory.last; // Delete to reset stuckness.
-        }
+    let returnCode = creep.moveTo(creep.targetPosObject, {
+      ignoreCreeps: true,
+      costCallback: creep.room.getCostMatrixCallback()
+    });
+
+    return true;
+  }
+  creep.say('Hi -> ' + creep.memory.search.target, true);
+  let returnCode = creep.move(creep.pos.getDirectionTo(search.path[0]));
+}
+
+roles.scout.breadthFirstSearch = function(creep) {
+  roles.scout.initSearch(creep);
+
+  (creep.memory.scoutSkip || creep.room.name === creep.memory.search.target)
+  && roles.scout.switchTarget(creep);
+
+  if (!creep.memory.search.target) {
+    creep.log('Suiciding: ' + JSON.stringify(creep.memory.search));
+    creep.suicide();
+    return true;
+  }
+
+  creep.targetPosObject = new RoomPosition(25, 25, creep.memory.search.target);
```

Comment:
pass this as separate parameter of `unStuckIt` and `move` functions would be much more clear then set custom property of game object